kolla/tests/test_build.py

```diff
@@ -57,15 +57,26 @@ def setUp(self):
         self.imageChild.parent = self.image
         self.imageChild.path = self.useFixture(fixtures.TempDir()).path
 
+    @mock.patch('docker.version', '2.7.0')
     @mock.patch.dict(os.environ, clear=True)
     @mock.patch('docker.APIClient')
-    def test_push_image(self, mock_client):
+    def test_push_image_before_v3_0_0(self, mock_client):
```

Comment:
What happens to versions between 2.7.0 and 3.0.0?

Reply:
They removed the command line option ```insecure_registry```. So when I run a task via Kolla, Kolla calls the python Docker module with the ```insecure_registry``` option, and it fails if the python docker module is >= 3.0.0. With so many different and independent libraries in use, it's quite common to bump into issues like this in OpenStack.

Reply:
What I meant was that `@mock.patch('docker.version', '2.7.0')` is explicitly defined and so is `@mock.patch('docker.version', '3.0.0')` so wasn't sure how the cases that lie between these range is handled.